app/robots/iCubGenova04/initVisualizer.m

```diff
@@ -2,8 +2,7 @@
 
 confVisualizer.robotName = robotName;
 confVisualizer.fileName = robot_config.fileName;
-confVisualizer.meshFilePrefix = robot_config.meshFilePrefix;
-confVisualizer.modelPath = robot_config.modelPath;
+[confVisualizer.modelPath,confVisualizer.meshFilePrefix] = wbs.getModelInfoFromRobotCfgAndYarpFinder(robot_config);
```

Comment:
This switches from getting the robotName from the `robotName` parameter to get it implicitly from the `YARP_ROBOT_NAME` env variable. Is this intentional? Should at least check that `robotName`  and `YARP_ROBOT_NAME` env variable match?

Reply:
As discussed F2F:
- `robotName` is set from the `YARP_ROBOT_NAME` variable in the init file, so they match.
- In the mindset of the initial approach, it would be safer to set it in the init file of the library (`initRobotDynamicsWithContactsCB.m`).
- As discussed F2F, it would be interesting to leave an option to the user to set the model file path (in the `init.m`) without having to use the Yarp machinery. But I just remembered that the WBToolbox blocks common configuration related to the model is set from the robot model found by the Yarp Resource Finder. If I'm not mistaken, the only inputs through the WBToolbox interface for the search are:
    - the robot file name `WBTConfigRobot.UrdfFile`. e.g. `'model.urdf'`
    - the `YARP_ROBOT_NAME`.

For that reason, for making sure the model information are coherent, I would go for only using one way to identify the model, through the YARP_ROBOT_NAME.

Reply:
Fixed in 63e2ccca5f91787bab333e6b51d92f31b8ba2a32.